cmake/linker/ld/target.cmake

```diff
@@ -52,6 +52,47 @@ macro(configure_linker_script linker_script_gen linker_pass_define)
   )
 endmacro()
 
+# Link a target to given libraries with toolchain-specific argument order
+#
+# Usage:
+#   toolchain_ld_link_elf(
+#     TARGET_ELF             <target_elf>
+#     LIBRARIES_PRE_SCRIPT   [libraries_pre_script]
+#     LINKER_SCRIPT          <linker_script>
+#     LIBRARIES_POST_SCRIPT  [libraries_post_script]
+#     DEPENDENCIES           [dependencies]
+#   )
+function(toolchain_ld_link_elf)
+  cmake_parse_arguments(
+    TOOLCHAIN_LD_LINK_ELF                                     # prefix of output variables
+    ""                                                        # list of names of the boolean arguments
+    "TARGET_ELF;LINKER_SCRIPT"                                # list of names of scalar arguments
+    "LIBRARIES_PRE_SCRIPT;LIBRARIES_POST_SCRIPT;DEPENDENCIES" # list of names of list arguments
+    ${ARGN}                                                   # input args to parse
+  )
+
+  target_link_libraries(
+    ${TOOLCHAIN_LD_LINK_ELF_TARGET_ELF}
+    ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_PRE_SCRIPT}
+    ${TOPT}
+    ${TOOLCHAIN_LD_LINK_ELF_LINKER_SCRIPT}
+    ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_POST_SCRIPT}
+
+    ${LINKERFLAGPREFIX},-Map=${PROJECT_BINARY_DIR}/${KERNEL_MAP_NAME}
+    -u_OffsetAbsSyms
+    -u_ConfigAbsSyms
```

Comment:
The symbols _ConfigAbsSyms and _OffsetAbsSyms are not toolchain specific and should therefore not be in toolchain-specific files.

We can do this for instance by doing something like this:

```
toolchain_ld_force_undefined_symbol(_OffsetAbsSyms _ConfigAbsSyms)

function(toolchain_ld_force_undefined_symbol)
foreach(flag ${ARGN})
zephyr_link_libraries(-u${flag})
endforeach()
endfunction()
```

This function could then be re-used, without the need to patch toolchain portability scripts, the next time we need to force a symbol to be undefined.

Reply:
Agree. Have tested and pushed your suggestion. 
I verified using strace that gnu ld command line is kept.